src/ripple/rpc/handlers/AccountInfo.cpp

```diff
@@ -120,48 +120,70 @@ Json::Value doAccountInfo (RPC::JsonContext& context)
             if (!txs.empty())
             {
                 jvQueueData[jss::txn_count] = static_cast<Json::UInt>(txs.size());
-                jvQueueData[jss::lowest_sequence] = txs.begin()->first;
-                jvQueueData[jss::highest_sequence] = txs.rbegin()->first;
-
                 auto& jvQueueTx = jvQueueData[jss::transactions];
                 jvQueueTx = Json::arrayValue;
 
+                std::uint32_t seqCount {0};
+                std::uint32_t ticketCount {0};
+                boost::optional<std::uint32_t> lowestSeq;
+                boost::optional<std::uint32_t> highestSeq;
+                boost::optional<std::uint32_t> lowestTicket;
+                boost::optional<std::uint32_t> highestTicket;
                 boost::optional<bool> anyAuthChanged(false);
                 boost::optional<XRPAmount> totalSpend(0);
 
-                for (auto const& [txSeq, txDetails] : txs)
+                for (auto const& tx : txs)
                 {
                     Json::Value jvTx = Json::objectValue;
 
-                    jvTx[jss::seq] = txSeq;
-                    jvTx[jss::fee_level] = to_string(txDetails.feeLevel);
-                    if (txDetails.lastValid)
-                        jvTx[jss::LastLedgerSequence] = *txDetails.lastValid;
-                    if (txDetails.consequences)
+                    if (tx.seqOrT.isSeq())
                     {
-                        jvTx[jss::fee] = to_string(
-                            txDetails.consequences->fee);
-                        auto spend = txDetails.consequences->potentialSpend +
-                            txDetails.consequences->fee;
-                        jvTx[jss::max_spend_drops] = to_string(spend);
-                        if (totalSpend)
-                            *totalSpend += spend;
-                        auto authChanged = txDetails.consequences->category ==
-                            TxConsequences::blocker;
-                        if (authChanged)
-                            anyAuthChanged.emplace(authChanged);
-                        jvTx[jss::auth_change] = authChanged;
+                        jvTx[jss::seq] = tx.seqOrT.value();
+                        seqCount += 1;
+                        if (!lowestSeq)
+                            lowestSeq = tx.seqOrT.value();
+                        highestSeq = tx.seqOrT.value();
```

Comment:
@cjcobb23, you're right.  It does rely on the order that transactions are retrieved by the `TxQ`.  There are a couple of ways to approach that:

1. Add a comment here.
2. Add a comment on `TxQ::getAccountTxs` describing the guaranteed return order.
3. Add a couple of asserts here (and a couple of tracking variables to help with the asserts).
4. Preemptively sort here in case the return order changes in the future.

I'm fine with any of those.  Preemptively sorting is maybe a little expensive, but the entry point is probably not called that often so I don't think it's prohibitive.  Do you have a preferred approach?  Thanks.

Reply:
@scottschurr I agree preemptive sorting might be a little expensive. I think adding a comment here and a couple of asserts would be sufficient.